[format.arg] Apply `\exposid` consistently

AlisdairM opened the pull request:
Consistently apply the `\exposid` macro when typesetting exposition-only private data members and typedef-name members in `basic_format_arg` and related parts.

Note that I confirmed the exposition-only intent against P0645R10 to ensure that the use of code font was a genuine misapplication of the original paper.

AlisdairM committed:
[format.arg] Apply `\exposid` consistently

Consistently apply the `\exposid` macro when typesetting exposition-only
private data members and typedef-name members in `basic_format_arg` and
related parts.

Note that I confirmed the exposition-only intent against P0645R10 to
ensure that the use of code font was a genuine misapplication of the
original paper.

jwakely commented on source/utilities.tex:
```diff
@@ -16511,8 +16511,8 @@
-    const void* ptr_;                                           // \expos
-    void (*format_)(basic_format_parse_context<char_type>&,
+    const void* @\exposid{ptr_}@;                                           // \expos
+    void (*@\exposid{format_}@)(basic_format_parse_context<char_type>&,
```
```suggestion
    void (*@\exposid{format_}@)(basic_format_parse_context<\exposid{char-type}>&,
```

jwakely replied:
More missed cases below in the definition of `handle` and its members.

jwakely replied:
Lines 16523, 16558, and 16570 (at least).

AlisdairM replied:
I believe I have now caught all missing usage of \exposid, but I thought that before too!

jwakely reviewed (changes requested):
I think you missed one, but otherwise I agree with the change. For libstdc++ I used this instead of the "not exposition-y" `char_type`:

    using _CharT = typename _Context::char_type;

JohelEGP commented on source/utilities.tex:
```diff
@@ -16675,7 +16675,7 @@
 \pnum
 \effects
-Initializes \tcode{size_} with \tcode{0}.
+Initializes \tcode{\exposid{size_}} with \tcode{0}.
```
```suggestion
Initializes \exposid{size_} with \tcode{0}.
```

JohelEGP commented on source/utilities.tex:
```diff
@@ -16688,8 +16688,8 @@
 Initializes
-\tcode{size_} with \tcode{sizeof...(Args)} and
-\tcode{data_} with \tcode{store.args.data()}.
+\tcode{\exposid{size_}} with \tcode{sizeof...(Args)} and
+\tcode{\exposid{data_}} with \tcode{store.args.data()}.
```
```suggestion
\exposid{size_} with \tcode{sizeof...(Args)} and
\exposid{data_} with \tcode{store.args.data()}.
```

AlisdairM committed:
Update source/utilities.tex

Co-authored-by: Jonathan Wakely <github@kayari.org>

AlisdairM committed:
Update source/utilities.tex

Thanks.  I made a very mechanical edit, and was not sure if the `\tcode` was redundant.

Co-authored-by: Johel Ernesto Guerrero Peña <johelegp@gmail.com>

AlisdairM committed:
Update source/utilities.tex

Co-authored-by: Johel Ernesto Guerrero Peña <johelegp@gmail.com>

AlisdairM committed:
Catch some missed exposid macros

This should catch all of the missing *char-type* exposid macros.

Additionally, add a trailing underscore to the exposition only
private data member, 'value', and consistently render it as
exposid as well.

JohelEGP commented on source/utilities.tex:
```diff
@@ -16304,7 +16304,7 @@
             float, double, long double,
             const @\exposid{char-type}@*, basic_string_view<@\exposid{char-type}@>,
-            const void*, handle> value;                                         // \expos
+            const void*, handle> @\exposid{value_}@;                                         // \expos
```
```suggestion
            const void*, handle> @\exposid{value_}@;                                        // \expos
```

AlisdairM replied:
Sorry, I don't see the difference.  Is this a parallel comment with the commit I just pushed?

JohelEGP replied:
No. You added a `_`, and I suggested removing a space before the `// expos` to keep it aligned.

AlisdairM committed:
Update source/utilities.tex

Co-authored-by: Johel Ernesto Guerrero Peña <johelegp@gmail.com>

jwakely reviewed (commented):
I don't think we have agreement on adding underscores to data member names, i.e. *`value`* vs *`value_`*. There is no `value()` member function that it could clash with, so *`value`* should be ok.

JohelEGP commented:
Underscores are discussed at https://github.com/cplusplus/draft/issues/4702.

jwakely commented:
> Underscores are discussed at #4702.

Indeed, and given the general dislike of the underscores and the strong objection, I don't think adding them here is progress. At the very least, controversial.

tkoeppe commented:
That's right, the naming of exposition-only variables is a delicate matter that we should not change as a drive-by.

AlisdairM commented:
While I strongly agree with the distaste for trailing underscores, I think this case merits special attention. `value` is the *only* exposition only private data member in the format framework that does *not* have a trailing underscore, unless it has internal dashes for kebab case to indicate that it is an exposition-only member.  See `args_`, `begin_`, `data_`, `end_`, `format_`, `indexing_`, `next_arg_id_`, `num_args_`, `out_`, `ptr_`, `separator_`, `size_`, `underlying_`, and even `opening-bracket_` and `closing-bracket_`.

On the other hand, I find just `str` in `basic_format_string` (so I was wrong about value being the _only_ example).

I suggest making `value_`, and possibly `str_`, consistent through this facility, and then tackling the whole issue if we want to deal with trailing underscores.

tkoeppe commented:
I see, yes, the rest of the formatter spec already uses underscores, so changing these two does indeed seem like an improvement. Go for it.

frederick-vs-ja commented:
I created a "manually rebased" version of this (#8971).
